src/internal/connector/exchange/api/contacts.go

```diff
@@ -55,6 +55,41 @@ func (c Contacts) CreateContactFolder(
 	return mdl, nil
 }
 
+// GetContactFolderByDisplayName fetches contact folder which has the displayName of
+// folderName. This is only done at root level.
+func (c Contacts) GetContactFolderByDisplayName(
+	ctx context.Context,
+	user, folderName string,
+) (models.ContactFolderable, error) {
+	filter := fmt.Sprintf("displayName eq '%s'", folderName)
+	options := &users.ItemContactFoldersRequestBuilderGetRequestConfiguration{
+		QueryParameters: &users.ItemContactFoldersRequestBuilderGetQueryParameters{
+			Filter: &filter,
+		},
+	}
+
+	resp, err := c.Stable.Client().UsersById(user).ContactFolders().Get(ctx, options)
+	if err != nil {
+		return nil, graph.Wrap(ctx, err, "getting contact folder").
+			With("folder name", folderName)
+	}
+
+	// We only expect one folder with provided folderName. Return an error if
```

Comment:
I think we could match within the result set, instead of discarding on n>1 responses.  It shouldn't require us to jump through that many hoops, we have some good aids for identification (such as the parent folder ID), and I think the resilience will pay off with our users. 

```suggestion
	// We only allow the api to match one folder with provided folderName. Return an error if
```

Reply:
Fair. I could make this function accept a parent folder id, and match against that in the result set. 
@ryanfkeepers  is there a convenient way to get the default contact folder id ( root)? I don't want to search by displayName.

Reply:
just found a solution for this - https://learn.microsoft.com/en-us/graph/known-issues#contacts

Reply:
oof, yeah, the default contacts folder is weird.  You might be able to find some other hacks we've used in the code as well (can't remember where they are, off the top of my head).

Or maybe this is something the container resolver could track for you?  Since it's aggregating all the containers, it should already know the root folder id.